### PR TITLE
Level Editor

### PR DESCRIPTION
**Pull request by bbeck13**
https://github.com/bbeck13/RhythmRunner/issues/40 This is the level editor.

### CODE REVIEW

**Comment by bbeck13** [comment]
One way to address https://github.com/bbeck13/RhythmRunner/issues/119 is to edit levels that are created from songs to make them playable if they are not

**Comment by bbeck13** [comment]
Added monsters but the obj file is messed up if some one has a working monster obj please replace it

**Comment by alexottoboni** [comment]
Is the monster file not consolidated into a single shape? I can fix that.

**Comment by bbeck13** [comment]
I'm not sure what's up with it take a look at the mesh and send me a fix if you can

File: src/game_updater/LevelEditorUpdater.cpp [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    game_state->GetPlayer()->SetPosition(game_state->GetCamera()->getLookAt());
+  }
+
+  if (ImGui::GetIO().KeysDown[GLFW_KEY_UP]) {
+    game_state->GetLevelEditorState()->SetCameraMove(camera_move * 1.2);
+    game_state->GetLevelEditorState()->SetCameraYawMove(camera_yaw_move * 1.2);
+  }
+  if (ImGui::GetIO().KeysDown[GLFW_KEY_DOWN]) {
+    game_state->GetLevelEditorState()->SetCameraMove(camera_move * 0.6);
+    game_state->GetLevelEditorState()->SetCameraYawMove(camera_yaw_move * 0.6);
```
Comment (Matthew-Stewart) [inline src/game_updater/LevelEditorUpdater.cpp:92]: *1.2 and *0.6 are different things (holding down will change the sensitivity basically twice as fast as holding up)

**Comment by josepharhar** [comment]
In order to make it compile on windows i had to add a target_link_libraries for LevelEditor. I put it on a branch named `jarhar.sfml.windoze`, if you like it you should add it to this pr

File: src/game_updater/LevelEditorUpdater.cpp (hunk 1 above)
Comment (bbeck13) [reply src/game_updater/LevelEditorUpdater.cpp:92]: ?? what should it be

**Comment by bbeck13** [comment]
Added windoze Cmake stuff

**Comment by josepharhar** [comment]
CR :pen: QA :pencil2:

**Pull request merged**